packages/ibm-products/src/components/CoachmarkFixed/CoachmarkFixed.js

```diff
@@ -0,0 +1,209 @@
+      className,
+      onClose = defaults.onClose,
+      portalTarget,
+      tagline = defaults.tagline,
+      theme = defaults.theme,
+      // Collect any other property values passed in.
+      ...rest
+    },
+    ref
+  ) => {
+    const overlayRef = useRef();
+    const portalNode = portalTarget
+      ? document.querySelector(portalTarget) ?? document.querySelector('body')
+      : document.querySelector('body');
+    const [isOpen, setIsOpen] = useState(false);
+    const [shouldResetPosition, setShouldResetPosition] = useState(false);
+    const [targetRect, setTargetRect] = useState();
+    const [targetOffset, setTargetOffset] = useState({ x: 0, y: 0 });
+    const [fixedIsVisible, setFixedIsVisible] = useState(false);
+
+    const handleClose = () => {
+      setFixedIsVisible(false);
+    };
+
+    const handleTransitionEnd = (e) => {
+      if (e.propertyName === 'transform' && !fixedIsVisible) {
+        setIsOpen(false);
+        onClose();
+      }
+    };
+
+    const handleTargetClick = (e) => {
+      setTargetRect(e.target.getBoundingClientRect());
+      setTargetOffset({ x: e.target.offsetLeft, y: e.target.offsetTop });
+      // reset position for all other kinds
+      setIsOpen(false);
+      setShouldResetPosition(true);
+    };
+
+    const contextValue = {
+      buttonProps: {
+        'aria-expanded': isOpen,
+        onClick: handleTargetClick,
+        // Compensate for accidental open/close on double-click.
+        // Only open on double-click.
+        onDoubleClick: handleTargetClick,
+      },
+      closeButtonProps: {
+        onClick: handleClose,
+      },
+      targetRect: targetRect,
+      targetOffset: targetOffset,
+      isOpen: isOpen,
+    };
+
+    // instead of toggling on/off,
+    // keep open and reset to original position
+    useEffect(() => {
+      if (shouldResetPosition) {
+        setShouldResetPosition(false);
+        setIsOpen(true);
+      }
+    }, [shouldResetPosition]);
+
+    useEffect(() => {
+      setFixedIsVisible(isOpen);
+    }, [isOpen]);
+
+    // On unmount:
+    // - DO NOT "Close()" the coachmark.
+    //   - This triggers a "signal" to close it forever.
+    //   - "Closing" should only ever be a user-triggered event.
+    // - DO "hide" the coachmark.
+    //   - The app is doing the action for the user.
+    //   - The user will have the opportunity to open it again.
+    useEffect(() => {
+      return () => setIsOpen(false);
+    }, []);
+
+    return (
+      <CoachmarkContext.Provider value={contextValue}>
+        <div
+          {
+            // Pass through any other property values as HTML attributes.
+            ...rest
+          }
+          className={cx(
+            blockClass, // Apply the block class to the main HTML element
+            `${blockClass}__${theme}`,
+            className, // Apply any supplied class names to the main HTML element.
+            // example: `${blockClass}__template-string-class-${kind}-n-${size}`,
```

Comment:
I think this could all be removed if you're no longer going to need to add these classes